### ISSUE TITLE
Look into docker run --init for drun

### ISSUE THREAD

**Issue by alblue** [alblue/scripts#3]
`docker run --init` executes a lightweight init script that intercepts signals and reaps processes. Running scripts with this flag enabled by default may make it easier to handle programs that fork or otherwise leave dead processes lying around. Will require testing against a bunch of container images for verification purposes to ensure that it doesn't introduce problems and that it forwards the return codes correctly.

**Comment by alblue** [comment]
It appears that exit codes are passed through as expected:

```
% docker run --rm -it ubuntu:20.04 bash
root@1480f08ad0ce:/# exit 127
exit
% echo $?
127
% docker run --rm -it --init ubuntu:20.04 bash
root@e2e1dfeebfdf:/# exit 127
exit
% echo $?
127

```

**Comment by alblue** [comment]
Tested on:

* ubuntu:{16.04,18.04,20.04,21.04}
* debian:{7,8,9,10}{,-slim}
* openjdk:{8,9,10,11,12,13,14,15,16,17}{,-alpine}
* rhel:{7,8}
* centos:centos{5,6,7,8}

No issues found, merged 795b4995940e64edd784053a7f4e639aa5e60a4d
